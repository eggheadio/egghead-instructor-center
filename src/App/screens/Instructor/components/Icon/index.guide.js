import React from 'react'
import {map, keys} from 'lodash'
import {storiesOf} from '@kadira/storybook'
import GuideVariation from '../../../../components/GuideVariation'
import Icon, {sizes, types} from './index'

storiesOf('Icon', module)

  .add('default', () => (
    <Icon type='more-info' />
  ))

  .add('sizes', () => (
    <div>
      {map(keys(sizes), size => (
        <GuideVariation tooltip={size}>
          <Icon
            type='more-info'
            size={size}
          />
        </GuideVariation>
      ))}
    </div>
  ))

  .add('types', () => (
    <div>
      {map(keys(types), type => (
        <GuideVariation tooltip={type}>
          <Icon type={type} />
        </GuideVariation>
      ))}
    </div>
  ))
