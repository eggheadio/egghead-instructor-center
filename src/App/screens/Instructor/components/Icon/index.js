import React, {PropTypes} from 'react'
import {keys} from 'lodash'

export const sizes = {
  1: '',
  2: 'fa-lg',
  3: 'fa-2x',
  4: 'fa-3x',
  5: 'fa-4x',
  6: 'fa-5x',
}

export const types = {
  'menu': 'bars',
  'step-incomplete': 'square-o',
  'step-complete': 'check-square-o',
  'close': 'close',
  'more-info': 'info-circle',
  'subscriber-minutes': 'clock-o',
  'revenue': 'money',
  'course': 'folder-open-o',
  'lesson': 'file-o',
}

const Icon = ({
  type,
  size = '1',
  className = '',
}) => (
  <span className={`
    fa
    fa-${types[type]} 
    ${sizes[size]}
    ${className}
  `} />
)

Icon.propTypes = {
  type: PropTypes.oneOf(keys(types)).isRequired,
  size: PropTypes.oneOf(keys(sizes)),
  className:  PropTypes.string,
}

export default Icon
