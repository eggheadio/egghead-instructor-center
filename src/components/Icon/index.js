// TODO: Consume from egghead-ui once published

import React, {PropTypes} from 'react'
import {keys} from 'lodash'

export const sizes = {
  1: '',
  2: 'fa-lg',
  3: 'fa-2x',
  4: 'fa-3x',
  5: 'fa-4x',
  6: 'fa-5x',
}

export const types = {
  'menu': 'bars',
  'step-incomplete': 'square-o',
  'step-complete': 'check-square-o',
  'close': 'close',
  'more-info': 'info-circle',
  'subscriber-minutes': 'clock-o',
  'revenue': 'money',
  'course': 'folder-open-o',
  'lesson': 'file-o',
  'refresh': 'refresh',
  'arrow-right': 'arrow-right',
}

const Icon = ({
  type,
  size = '1',
  spin = false,
  className = '',
}) => (
  <span className={`
    fa
    fa-${types[type]} 
    ${sizes[size]}
    ${className}
    ${spin ? 'fa-spin' : ''}
  `} />
)

Icon.propTypes = {
  type: PropTypes.oneOf(keys(types)).isRequired,
  size: PropTypes.oneOf(keys(sizes)),
  spin: PropTypes.bool,
  className:  PropTypes.string,
}

export default Icon
