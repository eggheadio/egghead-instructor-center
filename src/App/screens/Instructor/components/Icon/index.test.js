import React from 'react'
import {create} from 'react-test-renderer'
import {forEach, keys} from 'lodash'
import Icon, {types, sizes} from '.'

test('types', () => {
  forEach(keys(types), (type) => (
    expect(create(
      <Icon type={type} />
    )).toMatchSnapshot()
  ))
})

test('sizes', () => {
  forEach(keys(sizes), (size) => (
    expect(create(
      <Icon
        type='more-info'
        size={size}
      />
    )).toMatchSnapshot()
  ))
})
