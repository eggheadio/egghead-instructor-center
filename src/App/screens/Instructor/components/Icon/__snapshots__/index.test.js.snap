exports[`test sizes 1`] = `
<span
  className="
    fa
    fa-info-circle 
    
    
  " />
`;

exports[`test sizes 2`] = `
<span
  className="
    fa
    fa-info-circle 
    fa-lg
    
  " />
`;

exports[`test sizes 3`] = `
<span
  className="
    fa
    fa-info-circle 
    fa-2x
    
  " />
`;

exports[`test sizes 4`] = `
<span
  className="
    fa
    fa-info-circle 
    fa-3x
    
  " />
`;

exports[`test sizes 5`] = `
<span
  className="
    fa
    fa-info-circle 
    fa-4x
    
  " />
`;

exports[`test sizes 6`] = `
<span
  className="
    fa
    fa-info-circle 
    fa-5x
    
  " />
`;

exports[`test types 1`] = `
<span
  className="
    fa
    fa-bars 
    
    
  " />
`;

exports[`test types 2`] = `
<span
  className="
    fa
    fa-square-o 
    
    
  " />
`;

exports[`test types 3`] = `
<span
  className="
    fa
    fa-check-square-o 
    
    
  " />
`;

exports[`test types 4`] = `
<span
  className="
    fa
    fa-close 
    
    
  " />
`;

exports[`test types 5`] = `
<span
  className="
    fa
    fa-info-circle 
    
    
  " />
`;

exports[`test types 6`] = `
<span
  className="
    fa
    fa-clock-o 
    
    
  " />
`;

exports[`test types 7`] = `
<span
  className="
    fa
    fa-money 
    
    
  " />
`;

exports[`test types 8`] = `
<span
  className="
    fa
    fa-folder-open-o 
    
    
  " />
`;

exports[`test types 9`] = `
<span
  className="
    fa
    fa-file-o 
    
    
  " />
`;
